test/verify/check-networking-firewall

```diff
@@ -83,11 +83,13 @@ class TestFirewall(NetworkCase):
         wait_unit_state(m, "firewalld", "active")
 
         b.wait_in_text("#networking-firewall-summary", "{} active zone".format(active_zones))
+        time.sleep(10) # give Cockpit time to settle
```

Comment:
uh, 10s is looong time, that is something even user can hit. How does it propagate itself? Is something broken if you toggle it too quickly?

What does "settling" involve? I assume some API calls? If so, then we should either:
1. Wait in UI until those calls finish - I guess each call has some visible effect on the page, otherwise we would not do them
2. Show spinner/disable controls... while stuff is in progress

Reply:
> What does "settling" involve? I assume some API calls? If so, then we should either:
> 
>     1. Wait in UI until those calls finish - I guess each call has some visible effect on the page, otherwise we would not do them
> 
>     2. Show spinner/disable controls... while stuff is in progress

Ideally, yes.  However, there is no indication on the networking page except for the number of active zones and the activity is mostly triggered by signals from firewalld, I think.  So there is no well defined "we are done with startup" point I guess.

Of course, with more work we can improve this (and I don't really like sleeps like this at all, either).

We need to wait for the getServices calls to finish after getZones has returned the desired number of active zones.  The "change" signal is actually only emitted after the getServices is finished, hmm. The page is re-rendered independently of that apparently, and thus this flake is a consequence of the rewrite to React (which probably has changed the page updating dramatrically).

So, yeah. :-) Let's try to do better.

Reply:
It turns out that we don't need to slow down the tests at all.  Our client is robust enough, except for one bug, which is now fixed.  Thanks for not letting me get away with this!